src/lib/snapper/snapshot.rb

```diff
@@ -0,0 +1,96 @@
+# encoding: utf-8
+
+# ------------------------------------------------------------------------------
+# Copyright (c) 2015 SUSE LLC. All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact Novell, Inc.
+#
+# To contact Novell about this file by physical or electronic mail, you may find
+# current contact information at www.novell.com.
+# ------------------------------------------------------------------------------
+
+require "yast"
+require_relative "snapshot_dbus"
+
+module Yast2
+  module Snapper
+    class Snapshot
+      
+      include Yast::Logger
+
+      attr_accessor :number, :type, :pre_number, :timestamp, :user, :description, :cleanup_algo, :user_data, :number, :config, :strategy
+
+      STRATEGIES = { :dbus => SnapshotDBus }
+      TYPES = [:pre, :post, :single]
+
+      def initialize(number, type, pre_number, timestamp, user, cleanup_algo, description, user_data, config = "root", strategy = SnapshotDBus)
+        @number = number
+        @type = type
+        @pre_number = pre_number
+        @timestamp = timestamp
+        @user = user
+        @cleanup_algo = cleanup_algo
+        @description = description
+        @user_data = user_data
+        @strategy = strategy.new(self) 
+        @config = config
+      end
```

Comment:
so in the end, base class can end up as instance of Struct with defined attributes. And also it looks for me, that all attributs is not needed. Also strategy is not strategy, but strategy_class as you instanciate here. Really missing documentation there.